### ISSUE TITLE
_sslverify signs CertificateRequests with the insecure md5 hash

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7979]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7979|
|Type|enhancement|
|Created|2015-07-23 15:23:43Z|
|Branch|https://github.com/twisted/twisted/tree/no-md5-certsigning-7979|

This causes the tests to fail on a modern FIPS-mode OpenSSL (Fedora 22). We should use sha256.

<details><summary>Searchable metadata</summary>

```
trac-id__7979 7979
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_no_md5_certsigning_7979 branches/no-md5-certsigning-7979
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1437665023805784 1437665023805784
changetime__1439093076443895 1439093076443895
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

According to https://support.globalsign.com/customer/portal/articles/1499561-sha-256-compatibility , we should use it by default, as it's supported by OpenSSL 0.9.8o+, which everyone really should be on.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

This also causes RHEL7 to fail.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [45382]) Branching to no-md5-certsigning-7979.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

This looks good, see https://buildbot.twistedmatrix.com/boxes-unsupported?branch=/branches/no-md5-certsigning-7979 for the RHEL 7 builds. It also makes F22 only have one failure.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @hawkowl|
|-|-|

I think this is OK to land as-is, because it is strictly an upgrade, but we should probably have a more in-depth discussion about this code at some point in the near future; there are a number of other problems with it which need to be addressed.  (It sort of does what it says on the tin but not in a way which is useful for security.)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45416]) Merge no-md5-certsigning-7979: Sign certificate requests with sha256 instead of md5

Author: hawkowl
Reviewer: glyph
Fixes: [#7979](https://github.com/twisted/twisted/issues/7979)
